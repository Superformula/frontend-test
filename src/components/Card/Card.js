import React from "react";
import PropTypes from "prop-types";
import "./Card.scss";

import Button from "../Button/Button";
import Heading from "../Heading/Heading";
import Image from "../Image/Image";
import MetadataWrapper from "../MetadataWrapper/MetadataWrapper";
import Rating from "../Rating/Rating";
import Spacer from "../Spacer/Spacer";

const Card = props => {
  const { category, image, name, rating, price, status } = props;

  return (
    <div className="card-wrapper">
      <div className="card-metadata-container">
        <Image source={image} alt={name}/>
        <Spacer size="medium" />

        {/* Restaurant Name */}
        <Heading level={4}>{name}</Heading>
        <Spacer size="small" />

        {/* Rating */}
        <Rating rating={rating} />
        <Spacer size="small" />

        {/* Metadata - Category, Price, Status (open or closed) */}
        <MetadataWrapper category={category} status={status} price={price} />
        <Spacer size="large" />
      </div>

      <div className="card-button-container">
        <Button>Learn More</Button>
      </div>
    </div>
  );
};

Card.protoTypes = {
  category: PropTypes.string,
  image: PropTypes.string,
  name: PropTypes.string,
  rating: PropTypes.number,
  price: PropTypes.string,
  status: PropTypes.bool
};

export default Card;
