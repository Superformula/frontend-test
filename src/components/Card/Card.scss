@import "../../scss/base/typography";

.card-wrapper {
  display: flex;
  flex-direction: column;

  .card-metadata-container {
    flex: 1 0 auto;
  }
}
