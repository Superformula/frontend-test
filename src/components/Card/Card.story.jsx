import React from 'react';
import { storiesOf } from '@storybook/react';

import {
  Card,
  GridItem,
  Heading,
} from '..';

storiesOf('Card', module)
  .add('Styles', () => (
    <GridItem p={2}>
      <Heading>Card component</Heading>
      <p>Use the buttonVariant property to specify a different button style.</p>
      <GridItem display="flex" flexWrap="wrap">
        <Card
          imageSrc="https://placekitten.com/1000/666"
          title="Are you kitten me?"
          rating={4.5}
          price="$$$$"
          closed={false}
          onClick={() => {}}
        />
        <Card
          imageSrc="https://placekitten.com/1000/666"
          title="Is this fur real?"
          rating={1.5}
          price="$$"
          closed
          buttonVariant="secondary"
        />
      </GridItem>
    </GridItem>
  ));
