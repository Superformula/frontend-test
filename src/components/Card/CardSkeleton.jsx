import React from 'react';
import Skeleton from 'react-skeleton-loader';

import { GridItem } from '..';

export default () => (
  <GridItem
    width={[1, 1 / 3, 1 / 4]}
    display="flex"
    p={[0, 2, 3]}
    mb={4}
    mt={3}
  >
    <GridItem
      width={1}
      display="flex"
      flexDirection="column"
    >
      <Skeleton width="100%" height="288px" widthRandomness={0} borderRadius="2px" />
      <GridItem>
        <Skeleton width="75%" height="24px" widthRandomness={0} borderRadius="2px" />
        <div className="test">
          <Skeleton width="16px" height="16px" widthRandomness={0} borderRadius="50%" />
          <Skeleton width="16px" height="16px" widthRandomness={0} borderRadius="50%" />
          <Skeleton width="16px" height="16px" widthRandomness={0} borderRadius="50%" />
          <Skeleton width="16px" height="16px" widthRandomness={0} borderRadius="50%" />
        </div>
      </GridItem>
      <GridItem
        mt={4}
      >
        <Skeleton width="100%" height="44px" widthRandomness={0} borderRadius="2px" />
      </GridItem>
    </GridItem>
  </GridItem>
);
