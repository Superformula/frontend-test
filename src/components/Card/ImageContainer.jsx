import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const ImageWrap = styled.div`
  width: 100%;
  height: 228px;
  overflow: hidden;
  cursor: ${props => props.onClick ? 'pointer' : 'default'};
  &:hover > div,
  &: focus > div {
    transform: scale(1.05);
  }
`;

const ImageContent = styled.div`
  background: url(${props => props.img}) center / cover no-repeat;
  width: 100%;
  height: 100%;
  transition: all cubic-bezier(0.4, 0.0, 0.2, 1) 0.25s;
`;

const ImageContainer = ({ img, onClick }) => (
  <ImageWrap onClick={onClick}>
    <ImageContent img={img} />
  </ImageWrap>
);

ImageContainer.defaultProps = {
  onClick: null,
};

ImageContainer.propTypes = {
  img: PropTypes.string.isRequired,
  onClick: PropTypes.func,
};

export default ImageContainer;
