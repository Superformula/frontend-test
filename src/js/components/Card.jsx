import PropTypes from 'prop-types';
import Rating from 'react-rating';
import React from 'react';

import OpenClosedIndicator from './OpenClosedIndicator.jsx';

function Card(props) {
  const {
      category,
      closed,
      image,
      name,
      price,
      rating,
      url
    } = props,
    emptySymbol = '/src/images/star-empty.png',
    fullSymbol = '/src/images/star-filled.png';

  return (
    <div className="card">
      <div className="card-image">
        <img src={ image } alt={ name } />
      </div>
      <p className="card-name">{ name }</p>
      <Rating
        readonly
        placeholderRating={ rating }
        emptySymbol={ <img src={ emptySymbol } className="icon" alt="" /> }
        placeholderSymbol={ <img src={ fullSymbol } className="icon" alt="" /> }
        fullSymbol={ <img src={ fullSymbol } className="icon" alt="" /> }
      />
      <div className="row card-details">
        <div className="col-8 collapsed">
          { category } â€¢ { price }
        </div>
        <div className="col-4 collapsed text-right">
          { closed }
          <OpenClosedIndicator closed={ closed } />
        </div>
      </div>
      <a href={ url } className="btn">Learn More</a>
    </div>
  );
}

Card.propTypes = {
  category: PropTypes.string.isRequired,
  closed: PropTypes.bool.isRequired,
  image: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  rating: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};

export default Card;
