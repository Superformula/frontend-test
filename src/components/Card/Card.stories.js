import React from "react";

import Card from "./Card";

export default {
  component: Card,
  title: "Card"
};

const cardProps = {
  category: "thai",
  // image: ,
  headline: "Restaurant 2",
  rating: 4,
  price: "$$$",
  status: true
};

export const aCard = () => (
  <Card
    category={cardProps.category}
    // image: ,
    headline={cardProps.headline}
    rating={cardProps.rating}
    price={cardProps.price}
    status={cardProps.status}
  />
);
