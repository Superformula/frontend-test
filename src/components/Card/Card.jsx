import React from 'react';
import PropTypes from 'prop-types';

import {
  StarRating,
  Button,
  SubHeading,
  GridItem,
  Small,
  Badge,
} from '..';
import ImageContainer from './ImageContainer';

const Card = ({
  imageSrc,
  title,
  rating,
  price,
  closed,
  category,
  onClick,
  buttonVariant,
}) => (
  <GridItem
    width={[1, 1 / 3, 1 / 4]}
    display="flex"
    p={[0, 2, 3]}
    mb={4}
    mt={3}
  >
    <GridItem
      width={1}
      display="flex"
      flexDirection="column"
      bg="white"
    >
      <ImageContainer img={imageSrc} onClick={onClick} />
      <GridItem>
        <SubHeading mt={2}>{title}</SubHeading>
        <StarRating rating={rating} />
      </GridItem>
      <GridItem mb={3} display="flex" alignItems="center">
        <GridItem width={2 / 3}>
          {category
          && (
            <Small color="black54" textTransform="uppercase">
              {category}
              <GridItem px={1} display="inline">â€¢</GridItem>
            </Small>
          )}
          {price && <Small color="black54">{price}</Small>}
        </GridItem>
        <GridItem width={1 / 3} display="flex" justifyContent="flex-end">
          <Badge statusColor={closed ? 'red' : 'green'}>{closed ? 'Closed' : 'Open Now'}</Badge>
        </GridItem>
      </GridItem>
      <GridItem
        display="flex"
        flex="1 0 auto"
        alignItems="flex-end"
        mt={1}
      >
        <Button variant={buttonVariant} block onClick={onClick}>Learn More</Button>
      </GridItem>
    </GridItem>
  </GridItem>
);

Card.defaultProps = {
  price: undefined,
  category: undefined,
  onClick: null,
  buttonVariant: 'primary',
};

Card.propTypes = {
  imageSrc: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  rating: PropTypes.number.isRequired,
  closed: PropTypes.bool.isRequired,
  price: PropTypes.string,
  category: PropTypes.string,
  onClick: PropTypes.func,
  buttonVariant: PropTypes.string,
};

export default Card;
