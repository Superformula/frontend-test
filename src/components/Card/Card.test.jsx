import React from 'react';
import { mountWithTheme } from '../../helpers/tests/withTheme';
import Card from './Card';

const defaultProps = {
  imageSrc: 'http://awesometest.com/image.jpg',
  title: 'This is awesome',
  rating: 5,
  price: '$$$$',
  closed: false,
  category: 'italian',
  onClick: jest.fn(),
};
const setUp = (props = {}) => mountWithTheme(<Card {...{ ...defaultProps, ...props }}>Primary</Card>).find('Card');
describe('Card Component', () => {
  let component;
  it('renders itself with a primary button correctly', () => {
    component = setUp({ buttonVariant: 'primary' });
    expect(component).toMatchSnapshot();
  });
  it('renders itself with a secondary button correctly', () => {
    component = setUp({ buttonVariant: 'secondary' });
    expect(component).toMatchSnapshot();
  });
  it('has a working onClick function', () => {
    component.find('button').simulate('click');
    expect(defaultProps.onClick).toHaveBeenCalled();
  });
});
