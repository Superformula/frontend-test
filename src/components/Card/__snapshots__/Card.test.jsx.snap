// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component renders itself with a primary button correctly 1`] = `
<Card
  buttonVariant="primary"
  category="italian"
  closed={false}
  imageSrc="http://awesometest.com/image.jpg"
  onClick={[MockFunction]}
  price="$$$$"
  rating={5}
  title="This is awesome"
>
  <styled.div
    display="flex"
    mb={4}
    mt={3}
    p={
      Array [
        0,
        2,
        3,
      ]
    }
    width={
      Array [
        1,
        0.3333333333333333,
        0.25,
      ]
    }
  >
    <div
      className="sc-iwsKbI grHAwk"
      display="flex"
      width={
        Array [
          1,
          0.3333333333333333,
          0.25,
        ]
      }
    >
      <styled.div
        bg="white"
        display="flex"
        flexDirection="column"
        width={1}
      >
        <div
          className="sc-iwsKbI cdOkXf"
          display="flex"
          width={1}
        >
          <ImageContainer
            img="http://awesometest.com/image.jpg"
            onClick={[MockFunction]}
          >
            <styled.div
              onClick={[MockFunction]}
            >
              <div
                className="sc-dxgOiQ fEVAoz"
                onClick={[MockFunction]}
              >
                <styled.div
                  img="http://awesometest.com/image.jpg"
                >
                  <div
                    className="sc-ckVGcZ dxIIZh"
                  />
                </styled.div>
              </div>
            </styled.div>
          </ImageContainer>
          <styled.div>
            <div
              className="sc-iwsKbI fqrKBv"
            >
              <styled.h3
                mt={2}
              >
                <h3
                  className="sc-EHOje hZoKoK"
                >
                  This is awesome
                </h3>
              </styled.h3>
              <StarRating
                count={5}
                rating={5}
              >
                <styled.div
                  my={2}
                >
                  <div
                    className="sc-iwsKbI bQYrWc"
                  >
                    <styled.div
                      key="star-1"
                    >
                      <div
                        className="sc-bdVaJa bgsLNb"
                      >
                        <SvgStarFull>
                          <svg
                            data-file-name="SvgStarFull"
                          />
                        </SvgStarFull>
                      </div>
                    </styled.div>
                    <styled.div
                      key="star-2"
                    >
                      <div
                        className="sc-bdVaJa bgsLNb"
                      >
                        <SvgStarFull>
                          <svg
                            data-file-name="SvgStarFull"
                          />
                        </SvgStarFull>
                      </div>
                    </styled.div>
                    <styled.div
                      key="star-3"
                    >
                      <div
                        className="sc-bdVaJa bgsLNb"
                      >
                        <SvgStarFull>
                          <svg
                            data-file-name="SvgStarFull"
                          />
                        </SvgStarFull>
                      </div>
                    </styled.div>
                    <styled.div
                      key="star-4"
                    >
                      <div
                        className="sc-bdVaJa bgsLNb"
                      >
                        <SvgStarFull>
                          <svg
                            data-file-name="SvgStarFull"
                          />
                        </SvgStarFull>
                      </div>
                    </styled.div>
                    <styled.div
                      key="star-5"
                    >
                      <div
                        className="sc-bdVaJa bgsLNb"
                      >
                        <SvgStarFull>
                          <svg
                            data-file-name="SvgStarFull"
                          />
                        </SvgStarFull>
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
              </StarRating>
            </div>
          </styled.div>
          <styled.div
            alignItems="center"
            display="flex"
            mb={3}
          >
            <div
              className="sc-iwsKbI hCqzdK"
              display="flex"
            >
              <styled.div
                width={0.6666666666666666}
              >
                <div
                  className="sc-iwsKbI dLuuMB"
                  width={0.6666666666666666}
                >
                  <styled.small
                    color="black54"
                    textTransform="uppercase"
                  >
                    <small
                      className="sc-gzVnrw itQAs"
                      color="black54"
                    >
                      italian
                      <styled.div
                        display="inline"
                        px={1}
                      >
                        <div
                          className="sc-iwsKbI bUIrNs"
                          display="inline"
                        >
                          â€¢
                        </div>
                      </styled.div>
                    </small>
                  </styled.small>
                  <styled.small
                    color="black54"
                  >
                    <small
                      className="sc-gzVnrw cLDJWr"
                      color="black54"
                    >
                      $$$$
                    </small>
                  </styled.small>
                </div>
              </styled.div>
              <styled.div
                display="flex"
                justifyContent="flex-end"
                width={0.3333333333333333}
              >
                <div
                  className="sc-iwsKbI hsDEgj"
                  display="flex"
                  width={0.3333333333333333}
                >
                  <Badge
                    statusColor="green"
                    textTransform="uppercase"
                  >
                    <styled.div
                      alignItems="center"
                      display="flex"
                    >
                      <div
                        className="sc-iwsKbI bMDsUw"
                        display="flex"
                      >
                        <styled.span
                          bg="green"
                          display="inline-block"
                          mr={1}
                        >
                          <span
                            className="sc-htpNat jeXzCj"
                            display="inline-block"
                          />
                        </styled.span>
                        <styled.small
                          color="black54"
                          textTransform="uppercase"
                        >
                          <small
                            className="sc-gzVnrw itQAs"
                            color="black54"
                          >
                            Open Now
                          </small>
                        </styled.small>
                      </div>
                    </styled.div>
                  </Badge>
                </div>
              </styled.div>
            </div>
          </styled.div>
          <styled.div
            alignItems="flex-end"
            display="flex"
            flex="1 0 auto"
            mt={1}
          >
            <div
              className="sc-iwsKbI bujbry"
              display="flex"
            >
              <styled.button
                block={true}
                onClick={[MockFunction]}
                secondary={false}
                variant="primary"
              >
                <button
                  className="sc-bwzfXH kOCFzd"
                  onClick={[MockFunction]}
                >
                  Learn More
                </button>
              </styled.button>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </div>
  </styled.div>
</Card>
`;

exports[`Card Component renders itself with a secondary button correctly 1`] = `
<Card
  buttonVariant="secondary"
  category="italian"
  closed={false}
  imageSrc="http://awesometest.com/image.jpg"
  onClick={[MockFunction]}
  price="$$$$"
  rating={5}
  title="This is awesome"
>
  <styled.div
    display="flex"
    mb={4}
    mt={3}
    p={
      Array [
        0,
        2,
        3,
      ]
    }
    width={
      Array [
        1,
        0.3333333333333333,
        0.25,
      ]
    }
  >
    <div
      className="sc-iwsKbI grHAwk"
      display="flex"
      width={
        Array [
          1,
          0.3333333333333333,
          0.25,
        ]
      }
    >
      <styled.div
        bg="white"
        display="flex"
        flexDirection="column"
        width={1}
      >
        <div
          className="sc-iwsKbI cdOkXf"
          display="flex"
          width={1}
        >
          <ImageContainer
            img="http://awesometest.com/image.jpg"
            onClick={[MockFunction]}
          >
            <styled.div
              onClick={[MockFunction]}
            >
              <div
                className="sc-dxgOiQ fEVAoz"
                onClick={[MockFunction]}
              >
                <styled.div
                  img="http://awesometest.com/image.jpg"
                >
                  <div
                    className="sc-ckVGcZ dxIIZh"
                  />
                </styled.div>
              </div>
            </styled.div>
          </ImageContainer>
          <styled.div>
            <div
              className="sc-iwsKbI fqrKBv"
            >
              <styled.h3
                mt={2}
              >
                <h3
                  className="sc-EHOje hZoKoK"
                >
                  This is awesome
                </h3>
              </styled.h3>
              <StarRating
                count={5}
                rating={5}
              >
                <styled.div
                  my={2}
                >
                  <div
                    className="sc-iwsKbI bQYrWc"
                  >
                    <styled.div
                      key="star-1"
                    >
                      <div
                        className="sc-bdVaJa bgsLNb"
                      >
                        <SvgStarFull>
                          <svg
                            data-file-name="SvgStarFull"
                          />
                        </SvgStarFull>
                      </div>
                    </styled.div>
                    <styled.div
                      key="star-2"
                    >
                      <div
                        className="sc-bdVaJa bgsLNb"
                      >
                        <SvgStarFull>
                          <svg
                            data-file-name="SvgStarFull"
                          />
                        </SvgStarFull>
                      </div>
                    </styled.div>
                    <styled.div
                      key="star-3"
                    >
                      <div
                        className="sc-bdVaJa bgsLNb"
                      >
                        <SvgStarFull>
                          <svg
                            data-file-name="SvgStarFull"
                          />
                        </SvgStarFull>
                      </div>
                    </styled.div>
                    <styled.div
                      key="star-4"
                    >
                      <div
                        className="sc-bdVaJa bgsLNb"
                      >
                        <SvgStarFull>
                          <svg
                            data-file-name="SvgStarFull"
                          />
                        </SvgStarFull>
                      </div>
                    </styled.div>
                    <styled.div
                      key="star-5"
                    >
                      <div
                        className="sc-bdVaJa bgsLNb"
                      >
                        <SvgStarFull>
                          <svg
                            data-file-name="SvgStarFull"
                          />
                        </SvgStarFull>
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
              </StarRating>
            </div>
          </styled.div>
          <styled.div
            alignItems="center"
            display="flex"
            mb={3}
          >
            <div
              className="sc-iwsKbI hCqzdK"
              display="flex"
            >
              <styled.div
                width={0.6666666666666666}
              >
                <div
                  className="sc-iwsKbI dLuuMB"
                  width={0.6666666666666666}
                >
                  <styled.small
                    color="black54"
                    textTransform="uppercase"
                  >
                    <small
                      className="sc-gzVnrw itQAs"
                      color="black54"
                    >
                      italian
                      <styled.div
                        display="inline"
                        px={1}
                      >
                        <div
                          className="sc-iwsKbI bUIrNs"
                          display="inline"
                        >
                          â€¢
                        </div>
                      </styled.div>
                    </small>
                  </styled.small>
                  <styled.small
                    color="black54"
                  >
                    <small
                      className="sc-gzVnrw cLDJWr"
                      color="black54"
                    >
                      $$$$
                    </small>
                  </styled.small>
                </div>
              </styled.div>
              <styled.div
                display="flex"
                justifyContent="flex-end"
                width={0.3333333333333333}
              >
                <div
                  className="sc-iwsKbI hsDEgj"
                  display="flex"
                  width={0.3333333333333333}
                >
                  <Badge
                    statusColor="green"
                    textTransform="uppercase"
                  >
                    <styled.div
                      alignItems="center"
                      display="flex"
                    >
                      <div
                        className="sc-iwsKbI bMDsUw"
                        display="flex"
                      >
                        <styled.span
                          bg="green"
                          display="inline-block"
                          mr={1}
                        >
                          <span
                            className="sc-htpNat jeXzCj"
                            display="inline-block"
                          />
                        </styled.span>
                        <styled.small
                          color="black54"
                          textTransform="uppercase"
                        >
                          <small
                            className="sc-gzVnrw itQAs"
                            color="black54"
                          >
                            Open Now
                          </small>
                        </styled.small>
                      </div>
                    </styled.div>
                  </Badge>
                </div>
              </styled.div>
            </div>
          </styled.div>
          <styled.div
            alignItems="flex-end"
            display="flex"
            flex="1 0 auto"
            mt={1}
          >
            <div
              className="sc-iwsKbI bujbry"
              display="flex"
            >
              <styled.button
                block={true}
                onClick={[MockFunction]}
                secondary={false}
                variant="secondary"
              >
                <button
                  className="sc-bwzfXH cQaPdt"
                  onClick={[MockFunction]}
                >
                  Learn More
                </button>
              </styled.button>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </div>
  </styled.div>
</Card>
`;
